src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -0,0 +1,378 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Autodesk.DesignScript.Interfaces;
+using Dynamo.Logging;
+using ICSharpCode.AvalonEdit.CodeCompletion;
+using IronPython.Runtime;
+using IronPython.Runtime.Types;
+using Microsoft.Scripting;
+using Microsoft.Scripting.Actions;
+using Microsoft.Scripting.Hosting;
+using PythonNodeModels;
+
+namespace Dynamo.Python
+{
+
+
+    //This class needs to stay here with a reference to IronPython for now, its use will be
+    //completely removed from Dynamo though - so the dependencey on Ironpython will ONLY be at compile time.
+    [Obsolete("Do Not Use! This class will be removed in a future version of Dynamo," +
+        "Instead Reference the completion providers for a specific scripting engine, which implement " +
+        "IExternalCodeCompletionProvider")]
+    /// <summary>
+    /// Provides code completion for the Python Editor
+    /// </summary>
+    public class IronPythonCompletionProvider : LogSourceBase
+    {
+        private IExternalCodeCompletionProviderCore providerImplementation;
+        private const string providerTypeName = "DSIronPython.IronPythonCodeCompletionProviderCore, DSIronPython";
+
+        #region Properties and fields
+
+        /// <summary>
+        /// The engine used for autocompletion.  This essentially keeps
+        /// track of the state of the editor, allowing access to variable types and
+        /// imported symbols.
+        /// </summary>
+        private ScriptEngine engine;
+        public ScriptEngine Engine
+        {
+            get { return (ScriptEngine)(providerImplementation as ILegacyPythonCompletionCore).Engine; }
+            set { (providerImplementation as ILegacyPythonCompletionCore).Engine = value; }
+        }
+
+        /// <summary>
+        /// The scope used by the engine.  This is where all the loaded symbols
+        /// are stored.  It's essentially an environment dictionary.
+        /// </summary>
+        private ScriptScope scope;
+        public ScriptScope Scope
+        {
+            get { return (ScriptScope)(providerImplementation as ILegacyPythonCompletionCore).Scope; }
+            set { (providerImplementation as ILegacyPythonCompletionCore).Scope = value; }
+        }
+
+        /// <summary>
+        /// Already discovered variable types
+        /// </summary>
+        public Dictionary<string, Type> VariableTypes
+        {
+            get => providerImplementation.VariableTypes;
+            set => providerImplementation.VariableTypes = value;
```

Comment:
Interesting. I guess we should be allowed to use this now that our build machines have been upgraded